contrib/ansible/roles/etcd/tasks/main.yml

```diff
@@ -8,6 +8,11 @@
     - restart etcd
   when: not is_atomic
 
+- name: Clean etcd data directory
+  shell: rm -rf {{ etcd_data_dir }}/*
+  notify:
+    - restart etcd
```

Comment:
If you cannot listen on all your peer URLs, you have to go through [disaster recovery process](https://github.com/coreos/etcd/blob/master/Documentation/admin_guide.md#disaster-recovery) today.

We are working on [online snapshot](https://github.com/coreos/etcd/issues/3123) and import data into etcd. The expected workflow is that you could ask for an online snapshot, and you could import it into new etcd. It may be helpful for your case.

We are also improving the disaster recovery process to make it easy to run. But it is almost impossible to update list member by member offline because the update of member info always goes through consensus.